src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -56,16 +56,25 @@ private BoundDeconstructionAssignmentOperator BindDeconstructionAssignment(CShar
 
             if ((object)boundRHS.Type == null)
             {
-                if (boundRHS.Kind == BoundKind.TupleLiteral && !isDeclaration)
+                if (boundRHS.Kind == BoundKind.TupleLiteral)
                 {
-                    // tuple literal without type such as `(null, null)`, let's fix it up by peeking at the LHS
-                    TypeSymbol lhsAsTuple = MakeTupleTypeFromDeconstructionLHS(checkedVariables, diagnostics, Compilation);
-                    boundRHS = GenerateConversionForAssignment(lhsAsTuple, boundRHS, diagnostics);
+                    // Let's fix the literal up by figuring out its type
+                    // For declarations, that means merging type information from the LHS and RHS
+                    // For assignments, only the LHS side matters since it is necessarily typed
+                    TypeSymbol lhsAsTuple = MakeMergedTupleType(checkedVariables, (BoundTupleLiteral)boundRHS, node, Compilation, diagnostics);
+                    if (lhsAsTuple != null)
```

Comment:
Thanks. I'm still not quite sure when that is useful. Is it for certain types only (such as Symbols)?